packages/shared/src/hooks/useContextMenu.ts

```diff
@@ -21,7 +22,9 @@ export default function useContextMenu({
   );
   const client = useQueryClient();
   const { show, hideAll } = useContexifyContextMenu({ id });
-  const { data: isOpen } = useQuery(key, () => client.getQueryData(key));
+  const { data: isOpen } = useQuery(key, () =>
+    client.getQueryData<boolean>(key),
+  );
```

Comment:
Thinking about this now, I think in react query 4 we actually need to return something here as a default.
Did it not complain about this?

Reply:
It did not surprisingly. Do you know the specific doc page about this so I can look it up?

Reply:
We just need to make sure it is not `undefined` https://tanstack.com/query/v4/docs/react/guides/migrating-to-react-query-4#undefined-is-an-illegal-cache-value-for-successful-queries

Reply:
Yep as Ante mentions, I did a fix in a couple of items already in my PR.
(think modal and toast for instance)

Reply:
Updated. Thank you both!